js/modules/k6/grpc/client.go

```diff
@@ -421,7 +510,43 @@ func (c *Client) parseConnectParams(raw map[string]interface{}) (connectParams,
 			if params.MaxSendSize < 0 {
 				return params, fmt.Errorf("invalid maxSendSize value: '%#v, it needs to be a positive integer", v)
 			}
+		case "tls":
+			var ok bool
+			params.TLS, ok = v.(map[string]interface{})
+			err := fmt.Errorf("invalid tls value: '%#v', tls needs cert, key, (optional) password, and (optionally) cacerts", v)
```

Comment:
I am not a fan of us doing an allocation of this error even for every time there is no error.

Looking at the code - this is just validating stuff so `buildTLSConfigFromMap` can do 80% of the work ... but not return those errors.

But I would argue we can just have `params.TLS` be the result of `buildTLSConfigFromMap`, call it here with `v` and do all the validation there. 

I am not certain if `params.TLS`/`params` goes to js in some way and that makes this not possible. But it doesn't seem like it.

Reply:
Agreed. Refactored.